doc/source/whatsnew/v2.1.0.rst

```diff
@@ -128,6 +128,45 @@ Also, note that :meth:`Categorical.map` implicitly has had its ``na_action`` set
 This has been deprecated and will :meth:`Categorical.map` in the future change the default
 to ``na_action=None``, like for all the other array types.
 
+.. _whatsnew_210.enhancements.new_stack:
+
+New implementation of :meth:`DataFrame.stack`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+pandas has reimplemented :meth:`DataFrame.stack`. To use the new implementation, pass the argument ``v3=True``. This will become the only option in pandas 3.0.
```

Comment:
We did something similar with `describe` for keyword `datetime_is_numeric` that went away in 2.0; is there a way for `v3` to be a little more descriptive of the behavior change? Otherwise maybe `future_stack`?

Reply:
I'm fine with, but not in love with, `future_stack`. Agreed it's more reader friendly than `v3`. I'll see if I can come up with something better or go with that.

Reply:
I went with future_stack